File: Search/Python-Search/BinarySearch.py
def binary_search(arr, a, low, high):

    # Repeat until low and high meet each other
    while low <= high:

        mid = low + (high - low)//2

        if arr[mid] == a:
            return mid

        elif array[mid] < a:
            low = mid + 1

        else:
            high = mid - 1

    return -1

File: Search/C/BinarySearch/BinarySearch.c
int BinarySearch(int arr[], int a, int low, int high)
{
    while (low <= high)
    {

        int mid = low + (high - low) / 2;

        if (arr[mid] == a)
        {
            return mid;
        }
        else if (arr[mid] < a)
        {
            low = mid + 1;
        }
        else
        {
            high = mid - 1;
        }
    }

    return -1;
}
